draft-rha-jose-hpke-encrypt.md

```diff
@@ -220,44 +220,86 @@ This example demonstrates the representaton of an encapsulated key as a JWK.
+~~~
+{: #integrated-encryption-compact align="left" title="Integrated Encryption with Compact Serialization"}
+
+In the above example, the JWE Protected Header value is: 
+
+~~~
+{
+  "alg": ""HPKE-IntEnc",
+  "enc": "HPKE-Base-P256-SHA256-AES128GCM",
+  "epk": {
+    "kty": "EK",
+    "ek": "BGNkjzt076bsRGj78aX5AzT_HEOJBbY9q2Zo_5e7tbK0aPqu4eT1WI16jvRlZqpMyqZfP-RwR4Jwtaz_8U9thXA"
+  }
+}
+~~~
+
+~~~
+{
+    "protected": "eyJhbGciOiJkaXIiLCJlbmMiOiJIUEtFLUJhc2UtUDI1Ni1TSEEyNTYtQUVTMTI4R0NNIiwiZXBrIjp7Imt0eSI6IkVLIiwiZWsiOiJCTzRFbGZXd0xKRDZWcERza3c5LWxWMm9OMDJ2U1FKTW55ZHk3enhvSVlKZ1kydk9taE44Q1BqSHdRM1NONkhTcnNHNElacVpHVUR3dExKZjBoeHFTWGsifX0",
+    "ciphertext": "1ATsw0jshqPrv8CFcm9Rem9Wfi1Ygv30sozlRTtNNzcaaZ828GqP0AXtqQ1Msv8YXI9XZqh81MK3QnlZ7pOBC1BP7j00J1rrHujdb3zvnOpmJg"
+}
+~~~
+{: #integrated-encryption-json align="left" title="Integrated Encryption with JSON Serialization"}
+
+In the above example, the JWE Protected Header value is: 
+
+~~~
+{
+  "alg": ""HPKE-IntEnc",
+  "enc": "HPKE-Base-P256-SHA256-AES128GCM",
+  "epk": {
+    "kty": "EK",
+    "ek": "BGNkjzt076bsRGj78aX5AzT_HEOJBbY9q2Zo_5e7tbK0aPqu4eT1WI16jvRlZqpMyqZfP-RwR4Jwtaz_8U9thXA"
+  }
+}
 ~~~
-{: #compact-example align="left" title="Integrated Encryption with Compact Serialization"}
 
 ### Key Encryption
 
-This mode supports more than one recipient and JSON serializations.
+This mode supports more than one recipient.
 
 HPKE is used to encrypt the	Content Encryption Key (CEK), and the resulting ciphertext is included in the JWE ciphertext. 
 The plaintext will be encrypted using the CEK as explained in Step 15 of Section 5.1 of {{RFC7516}}.
 
-The sender MUST place the 'epk' and 'alg' parameters in the per-recipient unprotected header to indicate the use of HPKE. The "enc" (Encryption Algorithm) Header Parameter MUST be present in the JWE Protected Header.	 		
+When there are multiple recipients, the sender MUST place the 'epk' parameter in the per-recipient unprotected header to indicate the use of HPKE. In this case, the 'enc' (Encryption Algorithm) Header Parameter MUST be a content encryption algorithm from JSON Web Signature and Encryption Algorithms in {{JOSE-IANA}}, and it MUST be present in the JWE Protected Header.
 
 In Key Encryption mode: 
-- The JWE Encrypted Key MUST be the base64url encoded 'ct' value.
-- The JWE Initialization Vector MUST be produced as described in { Section 5.1 of RFC7516 }
-- The JWE Authentication Tag MUST be produced as described in { Section 5.1 of RFC7516 }
+
+*  The JWE Encrypted Key MUST be the base64url encoded 'ct' value.
+
+*  The JWE Initialization Vector MUST be produced as described in { Section 5.1 of RFC7516 }.
+
+*  The JWE Authentication Tag MUST be produced as described in { Section 5.1 of RFC7516 }.
```

Comment:
```suggestion
*  The JWE Encrypted Key MUST be the resulting HPKE ciphertext and tag ('ct' value) encoded using base64url.
```

These are clear, why do they need to be present?

Reply:
HPKE uses AEAD and it does not produce ciphertext and tag separately (the tag is appended to the ciphertext).

Reply:
I'm aware, but this is describing Key Encryption, not Integrated Encryption. In the conceptually similar Key Agreement with Key Wrapping the KW's iv and tag would go into a header and not be the JWE IV or Tag. So this does not need to be called out. 

The fact that they're absent is called in Integrated Encryption correctly but in here there's nothing that needs to be said.

Reply:
Yes, I removed the text related to JWE IV and Authentication Tag.